### PR TITLE
Webconfig optimization on tmp cache deletion to avoid concurrency issues

### PR DESCRIPTION
**Pull request by Sreedevi29**

### CODE REVIEW

File: src/webcfg_pack.c [hunk 1]
```diff
@@ -100,8 +100,12 @@ ssize_t webcfgdb_blob_pack(webconfig_db_data_t *webcfgdb, webconfig_tmp_data_t *
 
     while(temp_data != NULL)
     {
-        temp_data = temp_data->next;
-        tmp_count++;
+	WebcfgDebug("The temp name is %s\n",temp_data->name);
+	if(strcmp("success", temp_data->status) != 0)
```
Comment (shilpa24balaji) [inline src/webcfg_pack.c:104]: Do we need to add status null check here?

File: src/webcfg_pack.c [hunk 2]
```diff
@@ -184,46 +188,48 @@ ssize_t webcfgdb_blob_pack(webconfig_db_data_t *webcfgdb, webconfig_tmp_data_t *
        {
 	    while(temp_data != NULL) //1 element
 	    {
-                msgpack_pack_map( &pk, 5); //name, version
-
-	        struct webcfg_token WEBCFG_MAP_TEMP_NAME;
+		if(strcmp("success", temp_data->status) != 0) //to avoid duplicate entries
```
Comment (shilpa24balaji) [inline src/webcfg_pack.c:191]: Do we need to add status null check here?

File: src/webcfg.c [hunk 3]
```diff
@@ -171,6 +171,22 @@ void *WebConfigMultipartTask(void *status)
 		{
 			if(maintenance_doc_sync == 1 && checkMaintenanceTimer() == 1 )
 			{
+				char *ForceSyncDoc = NULL;
+				char* ForceSyncTransID = NULL;
+				getForceSync(&ForceSyncDoc, &ForceSyncTransID);
+				if((ForceSyncDoc == NULL) && (ForceSyncTransID == NULL) && (!forced_sync) && (!get_bootSync()))
+				{
+					WebcfgDebug("release success docs at every maintenance window\n");
```
Comment (sadhyama) [inline src/webcfg.c:179]: we can enable this log as INFO .

**Comment by Sreedevi29** [comment]
Old PR link https://github.com/xmidt-org/webcfg/pull/65/ to get the continuation.

**Review by sadhyama — APPROVED** [review]

**Pull request merged**